initrd-progs/0initrd/sbin/mountpartition

```diff
@@ -29,15 +29,6 @@ mkdir -p ${MNT_DIR}
 
 #-----------------------------------------------------
 
-. /etc/rc.d/functions_x
-MNT_O='noatime'
-MNT_DSK=$(fx_get_drvname $MNT_DEV)
-if [ "$MNT_FS" = "ext4" -o "$MNT_FS" = "f2fs" ] ; then
-  if [ "$(cat /sys/block/${MNT_DSK}/queue/rotational 2>/dev/null)" = "0" ] ; then
-    MNT_O="-o discard,noatime" # enable SSD TRIM
```

Comment:
How does this even work? If line 37 works, then line 33 is broken. If I understand correctly, later in this file we pass `-o $MNT_O` to `mount`, and `-o -o` doesn't work.

Reply:
Amazing, now I understand why that extra `-o` is not a problem: this piece of code is broken since ea50d51e87. The condition is always false, so the value of MNT_O is never changed to the bad value that won't work.

initrd-progs/0initrd/init has this line, which passes `/dev/whatever` and not `whatever` to mountpartition:

```
 /sbin/mountpartition /dev/${1} $ONE_MP $ONE_FS #-t $ONE_FS /dev/$ONE_PART $ONE_MP
```

Then, mountpartition puts the first argument in MNT_DEV and passes it to fx_get_drvname:

```
MNT_DEV="$1"  #ex: /dev/sda1

...

xdev=${MNT_DEV##*/} #basename

...

MNT_DEV=/dev/$xdev

...

MNT_DSK=$(fx_get_drvname $MNT_DEV)

...

  if [ "$(cat /sys/block/${MNT_DSK}/queue/rotational 2>/dev/null)" = "0" ] ; then

...
```

This `if` is always false, because fx_get_drvname expects `sdb1`, not `/dev/sdb1`. It returns `de` and `/sys/block/de/queue/rotational` doesn't exist. Therefore, TRIM is never enabled.

```
~$ MNT_DEV=/dev/sdb1
+ MNT_DEV=/dev/sdb1
~$ xdev=${MNT_DEV##*/} #basename
+ xdev=sdb1
~$ MNT_DEV=/dev/$xdev
+ MNT_DEV=/dev/sdb1
~$ MNT_DSK=$(fx_get_drvname $MNT_DEV)
++ fx_get_drvname /dev/sdb1
++ case $1 in
++ echo /de
+ MNT_DSK=/de
~$cat /sys/block/${MNT_DSK}/queue/rotational
+ cat /sys/block//de/queue/rotational
cat: /sys/block//de/queue/rotational: No such file or directory
```